src/messageReactionAdd.ts

```diff
@@ -1,45 +1,84 @@
-import { MessageReaction } from 'discord.js';
+import { MessageReaction, User, PartialUser } from 'discord.js';
 
 import config from './config';
+import { log } from './logger';
 
 const { REACTIONS_COUNT, ROLE } = config;
 
-const getTotalReactionCount = async (reaction: MessageReaction) => {
+const getTotalReactionCount = async (reaction: MessageReaction, user: User | PartialUser) => {
     // Map through each reaction emoji and return the count
     const getReactionCounts = reaction.message.reactions.cache.map(async (value) => {
         if (value.partial) {
             try {
                 await value.fetch();
             } catch (error) {
-                return Promise.resolve(0)
+                return Promise.resolve(0);
             }
         }
 
-        return Promise.resolve(value.count!)
-    })
+        // ignore reactions from the message author
+        if (user.id === reaction.message.author.id) {
```

Comment:
If the `user.id` is the reacted user who triggered the event then you are comparing the author id to the latest user who reacted instead of checking every user reacted to the message.

I think a better approach to solve this issue is to get the users who reacted to the message and filter out the author and count them.

Reply:
OMG you are so right 🤦‍♂️  , another great spot 👍

Reply:
I think the latest change is better now - it was hard to test with just me though 🤞